### PR TITLE
Integrate `infection/infection` to improve tests quality

### PR DESCRIPTION
**Pull request by maks-rafalko**
Depends on https://github.com/sidz/phpstan-rules/pull/7 and https://github.com/sidz/phpstan-rules/pull/8

Currently in Draft, to trigger CI and see if it works in some way.

- [x] Check if CI is correctly triggered
- [x] Fix all the escaped mutants
- [ ] Discuss if we need to create a separate `mt.yaml` to run it only for one PHP version

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -56,3 +56,7 @@ jobs:
       - name: Run Unit tests
         shell: bash
         run: make tests
+
+      - name: Run Infection
+        shell: bash
+        run: make infection
```
Comment (maks-rafalko) [inline .github/workflows/ci.yaml:62]: instead, try running it with `--git-diff-lines`

**Comment by sidz** [comment]
Could you please rebase against latest master and resolve conflicts?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (maks-rafalko) [reply .github/workflows/ci.yaml]: done

**Comment by maks-rafalko** [comment]
@sidz ready for review

Here you can see Infection takes almost 2m when executed for the whole project: https://github.com/sidz/phpstan-rules/actions/runs/5126088310/jobs/9220118095#step:9:1

After running it for the changed lines only, it takes 1s (for this particular PR because no source files are changed): https://github.com/sidz/phpstan-rules/actions/runs/5126162099/jobs/9220300054#step:9:1

But anyway, this is a good improvement.

Regarding MSI, it's now 100%

**Comment by sidz** [comment]
> Discuss if we need to create a separate mt.yaml to run it only for one PHP version

Fine for me to leave it as is.

**Comment by sidz** [comment]
Thank you @maks-rafalko

**Pull request merged**